src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotContainUnderscores.Fixer.cs

```diff
@@ -19,11 +25,33 @@ public sealed override FixAllProvider GetFixAllProvider()
             return WellKnownFixAllProviders.BatchFixer;
         }
 
-        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // Fixer not yet implemented.
-            return Task.CompletedTask;
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var model = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
 
+            var node = root.FindNode(context.Span);
+            var symbol = model.GetDeclaredSymbol(node, context.CancellationToken);
+
+            string title = MicrosoftCodeQualityAnalyzersResources.IdentifiersShouldNotContainUnderscoresTitle;
+            context.RegisterCodeFix(new MyCodeAction(title,
+                                        async ct => await RenameWithoutUnderscores(context.Document, symbol, ct).ConfigureAwait(false),
+                                        equivalenceKey: title),
+                                    context.Diagnostics);
+        }
+
+        private static async Task<Document> RenameWithoutUnderscores(Document document, ISymbol symbol, CancellationToken cancellationToken)
+        {
+            Solution newSolution = await Renamer.RenameSymbolAsync(document.Project.Solution, symbol, symbol.Name.Replace("_", ""), null, cancellationToken).ConfigureAwait(false);
+            return newSolution.GetDocument(document.Id);
```

Comment:
This is written just like [this](https://github.com/dotnet/roslyn-analyzers/blob/0cbdd68e14722e911d50eee64ee95823b3ffa09c/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/ParameterNamesShouldMatchBaseDeclaration.Fixer.cs#L59).

But I'm getting a null-reference warning:

![image](https://user-images.githubusercontent.com/31348972/83555288-5ad11b00-a50e-11ea-9449-8882a59869a6.png)

Why a warning is emitted here but not there? I can't get the difference.

Reply:
GetDocument can return a null value if the document ID does not belong to the solution. You should add a `!` suppression here. We should consider adding GetRequiredXXX extension methods to our repo similar to the ones in Roslyn: https://github.com/dotnet/roslyn/blob/bc2d6a2597d07a574be4e292f9dbd9c8f975fd7f/src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/ISolutionExtensions.cs#L36-L68

Reply:
@mavasani Thanks. I didn't see the `!` there 😆 

That makes sense now.

Reply:
Should I add the extension methods in this PR? or leave it as another task?

A small question: why the `ISolutionExtensions` is made internal in roslyn? I can't see an obvious problem with exposing these extensions to be public.

Reply:
I think there is possible a pending work item to make these public with API review approval. @CyrusNajmabadi do you know?

Reply:
> Should I add the extension methods in this PR? or leave it as another task?

A separate PR might get in quicker.

Reply:
> I think there is possible a pending work item to make these public with API review approval. @CyrusNajmabadi do you know?

Don't know what we're planning there.  It's a pretty awful situation all around.  i'd prefer we just break these to throw.